de.peeeq.wurstscript/src/main/java/de/peeeq/wurstscript/intermediatelang/optimizer/SimpleRewrites.java

```diff
@@ -189,6 +191,12 @@ private void optimizeOpCall(ImOperatorCall opc) {
                 } else {
                     wasViable = false;
                 }
+            } else if (left.structuralEquals(right) && opc.getOp() == WurstOperator.PLUS && !(left instanceof ImStringVal)) {
+                if (!(left instanceof ImFunctionCall && sideEffectAnalysis.hasSideEffects((ImFunctionCall) left))) {
```

Comment:
You should check for side-effects in general. Not only `ImFunctionCall`s can have side effects. There's also statement expressions, and expressions with side-effects nested in other expressios. For example `(1 + f()) + (1 + f())`.